### ISSUE TITLE
Failure on volume directory existing

### ISSUE THREAD

**Issue by vbatts** [projectatomic/buildah#439]
**Description**

**Steps to reproduce the issue:**
1. git clone https://github.com/freeipa/freeipa-container.git
2. freeipa-container
3. sudo buildah bud -t freeipa .

**Describe the results you received:**
```shell
[...]
/var/named/dynamic/
/var/named/dyndb-ldap/
STEP 32: RUN rm -rf /var/log-removed
STEP 33: RUN sed -i 's!^d /var/log.*!L /var/log - - - - /data/var/log!' /usr/lib/tmpfiles.d/var.conf
STEP 34: RUN mv /usr/lib/tmpfiles.d/journal-nocow.conf /usr/lib/tmpfiles.d/journal-nocow.conf.disabled
STEP 35: RUN rm -f /data-template/var/lib/systemd/random-seed
STEP 36: RUN echo 1.1 > /etc/volume-version
STEP 37: ENV container docker
STEP 38: EXPOSE 53/udp 53 80 443 389 636 88 464 88/udp 464/udp 123/udp 7389 9443 9444 9445
STEP 39: VOLUME [ "/tmp", "/run", "/data", "/var/log/journal" ]
error building: error building at step {Env:[container=docker DISTTAG=f26container FGC=f26   PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin container=docker] Command:volume Args:[/tmp /run /data /var/log/journal] Flags:[] Attrs:map[json:true] Message:VOLUME /tmp /run /data /var/log/journal Original:VOLUME [ "/tmp", "/run", "/data", "/var/log/journal" ]}: error ensuring volume path "/var/lib/containers/storage/overlay/7168d336ffed21a31a8ee96d59a5d1b34a4073082f7cfcac165d868a389c48fd/merged/var/log/journal" exists: mkdir /var/lib/containers/storage/overlay/7168d336ffed21a31a8ee96d59a5d1b34a4073082f7cfcac165d868a389c48fd/merged/var/log: file exists
```

**Describe the results you expected:**
a built image

**Output of `rpm -q buildah` or `apt list buildah`:**

```
buildah-0.11-1.git6bad262.fc26.x86_64
```

**Output of `buildah version`:**

```
Version:       0.11
Go Version:    go1.8.5
Image Spec:    1.0.0
Runtime Spec:  1.0.0
Git Commit:    6bad262
Built:         Wed Jan 17 08:49:09 2018
OS/Arch:       linux/amd64
```

**Output of `cat /etc/*release`:**

```
Fedora release 26 (Twenty Six)
NAME=Fedora
VERSION="26 (Workstation Edition)"
ID=fedora
VERSION_ID=26
PRETTY_NAME="Fedora 26 (Workstation Edition)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:26"
HOME_URL="https://fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=26
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=26
PRIVACY_POLICY_URL=https://fedoraproject.org/wiki/Legal:PrivacyPolicy
VARIANT="Workstation Edition"
VARIANT_ID=workstation
Fedora release 26 (Twenty Six)
Fedora release 26 (Twenty Six)
```

**Output of `uname -a`:**

```
Linux chacha.usersys.redhat.com 4.14.14-200.fc26.x86_64 #1 SMP Fri Jan 19 13:27:06 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
(paste your output here)
```

**Comment by rhatdan** [comment]
Urvahi can you take a look at this?

**Comment by umohnani8** [comment]
@rhatdan will do.

**Comment by adelton** [comment]
Note, https://github.com/freeipa/freeipa-container/pull/188 should fix the specific freeipa-container issue.

**Comment by adelton** [comment]
The minimal reproducer which triggers this issue and shows the difference between `docker build` and `buildah build-using-dockerfile` would be

```
FROM registry.fedoraproject.org/fedora:26
RUN mkdir -p /data
RUN ln -s /data/log /test-log
VOLUME [ "/test-log/test" ]
```

**Comment by adelton** [comment]
So the problem seems to be that `buildah` cannot walk symlinks.

**Comment by rhatdan** [comment]
@umohnani8 Did you ever get anywhere with this issue?

**Comment by umohnani8** [comment]
@rhatdan working on it.

**Comment by umohnani8** [comment]
Fixed in #541
